i2c/ssd1306_i2c/ssd1306_i2c.c

```diff
@@ -387,13 +387,23 @@ int main() {
         "by the name of",
         "    PICO"
     };
-
     int y = 0;
-    for (uint i = 0 ;i < count_of(text); i++) {
+    for (uint i = 0; i < count_of(text); i++) {
         WriteString(buf, 5, y, text[i]);
-        y+=8;
+        y += 8;
+        // Height limit reached. Show some lines.
+        if (y == SSD1306_HEIGHT) {
+            render(buf, &frame_area);
+            sleep_ms(3000);
```

Comment:
So what happens when you run this app is:

1. Lines 0 to 3 get printed
1. Wait 3s
1. Lines 4 to 7 get printed
1. Wait 3s
1. The extra check on lines 403 to 406 doesn't run with the default text because y is 0
1. Wait another 3s (this is the extra sleep on line 409 we should remove)

Leaving the sleep on line 397 and 405 ensures that text always (and only) displays for exactly 3s

Reply:
Sounds good!